SlaveWAMPServer.spec.js

```diff
@@ -161,60 +167,247 @@ describe('SlaveWAMPServer', () => {
 			})).to.be.ok();
 
 			expect(workerMock.sendToMaster.called).not.to.be.ok();
 		});
 	});
+
+	describe('sendToMaster', () => {
+		before(() => {
+			sinon.stub(Math, 'random').returns(0);
+		});
+
+		after(() => {
+			Math.random.restore();
+		});
+
+		it('should pass correct InterProcessRPCRequestSchema compatible request to sendToMaster function', () => {
+			slaveWAMPServer.sendToMaster(validProcedure, validData, validSocketId, validCb);
+			expect(workerMock.sendToMaster.calledOnce).to.be.true();
+			expect(workerMock.sendToMaster.calledWith({
+				type: '/InterProcessRPCRequestSchema',
+				procedure: validProcedure,
+				data: validData,
+				socketId: validSocketId,
+				workerId: 0,
+				signature: validSignature,
+			})).to.be.ok();
+		});
+
+		it('should pass create a new entry in interProcessRPC map', () => {
+			slaveWAMPServer.sendToMaster(validProcedure, validData, validSocketId, validCb);
+			expect(slaveWAMPServer.interProcessRPC).to.have.nested.property(`${validSocketId}.${validProcedure}.${validSignature}`).to.be.a('function');
+		});
+	});
+
+	describe('onSocketDisconnect', () => {
+		it('should leave interProcessRPC in initial state when invoked without arguments', () => {
+			slaveWAMPServer.onSocketDisconnect();
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should leave interProcessRPC in initial state when entry does not exist', () => {
+			slaveWAMPServer.onSocketDisconnect(validSocketId);
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should remove existing entry from interProcessRPC', () => {
+			slaveWAMPServer.interProcessRPC = validInterProcessRPCEntry;
+			slaveWAMPServer.onSocketDisconnect(validSocketId);
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+	});
+
+	describe('saveCall', () => {
+		it('should throw an error when invoked without arguments', () => {
+			expect(slaveWAMPServer.saveCall).to.throw('Cannot save a call for wrong InterProcessRPCRequest request');
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without socket id', () => {
+			delete validRequest.socketId;
+			expect(() => {
+				slaveWAMPServer.saveCall(validRequest, validCb);
+			}).to.throw('Cannot save a call for wrong InterProcessRPCRequest request');
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without procedure', () => {
+			delete validRequest.procedure;
+			expect(() => {
+				slaveWAMPServer.saveCall(validRequest, validCb);
+			}).to.throw('Cannot save a call for wrong InterProcessRPCRequest request');
+			expect(slaveWAMPServer.interProcessRPC).to.be.empty();
+		});
+
+		it('should throw an error when invoked without signature', () => {
+			delete validRequest.signature;
+			expect(() => {
+				slaveWAMPServer.saveCall(validRequest, validCb);
+			}).to.throw('Cannot save a call for wrong InterProcessRPCRequest request');
```

Comment:
It's difficult to see the relationship between these errors without storing it in a common variable.